File: codeforce007/day-2/Twins.cpp
#include <iostream>
#include <bits/stdc++.h>
#include <vector>
using namespace std;

int main(){
  int n,minCoins,totalSum,partialCoinSum;
  minCoins=totalSum=partialCoinSum=0;
  vector<int> v;
  cin >>n;
  for(int i=0;i<n;i++){
    int temp;
    cin >>temp;
    v.push_back(temp);
    sort(v.begin(),v.end());
    totalSum+=temp;
  }
  for(int i=n-1;i>=0;i--){
    partialCoinSum+=v[i];
    minCoins+=1;
    if(partialCoinSum>(totalSum-partialCoinSum)){
      cout<<minCoins;
      break;
    }
  }
  return 0;
}

File: codeforce007/day-2/Twins.py
n = int(input())
coins = [int(i) for i in input().split()]
coins.sort(reverse=True)
totalSum=0
minCoins=0
coinSum=0
for i in coins:
    totalSum+=i

for i in coins:
    coinSum+=i
    minCoins+=1
    if coinSum>totalSum-coinSum:
        print(minCoins)
        break
